proxy/protocol.go

```diff
@@ -128,19 +128,20 @@ func (proto *protocol) Serve(conn net.Conn, userData interface{}) error {
 		// Execute the corresponding handler
 		resp := proto.handleRequest(ctx, &req, &hr)
 
-		// First send an fd if the handler associated a file with the response
+		// Send the response back to the client.
+		if err = api.WriteMessage(conn, resp); err != nil {
+			// Something made us unable to write the response back
+			// to the client (could be a disconnection, ...).
+			return err
+		}
+
+		// And send a fd if the handler associated a file with the response
 		if hr.file != nil {
 			if err = api.WriteFd(conn.(*net.UnixConn), int(hr.file.Fd())); err != nil {
 				return err
 			}
 			hr.file.Close()
```

Comment:
Not directly related to your patch, but as you are in the area...
It looks like if we err on the WriteFd(), then we return before the hd.file.Close() - which looks like a potential file handle leak?
I note there is no file.Close() in any of the other error paths either - it feels a little odd and inconsistent. Any thoughts (though this should not block this PR, as that code was untouched I think)